pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -656,21 +666,21 @@ func (jm *ControllerV2) cleanupFinishedJobs(ctx context.Context, cj *batchv1.Cro
 		}
 	}
 
-	if cj.Spec.SuccessfulJobsHistoryLimit != nil {
+	if cj.Spec.SuccessfulJobsHistoryLimit != nil &&
 		jm.removeOldestJobs(cj,
```

Comment:
is it possible that this function doesn't remove any jobs?

Reply:
Yes, it's possible. I have fixed it.